### PR TITLE
Milestone 3

### PR DESCRIPTION
**Pull request by gitnyasha**
- Made the BooksForm component stateful
- Connected the REMOVE_BOOK action to the component

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -35,20 +27,17 @@
     "import/extensions": "off",
     "react/jsx-uses-vars": "error",
     "react/jsx-uses-react": "error",
+    "react/prop-types": 0,
+    "react/destructuring-assignment": [0, { "extensions": [".jsx"] }],
+    "treatUndefinedAsUnspecified": 0,
     "quotes": [
```
Comment (bellom) [inline .eslintrc.json:33]: Opps, we have some little changes in this file, check out the [microverse-linter-config](https://github.com/microverseinc/linters-config/blob/master/react-redux/.eslintrc.json)

image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/

File: src/components/Book.js [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
+      <td>
+        <button
+          type="submit"
+          onClick={() => {
+            removeBook(book);
+          }}
+        >
+          Remove
+        </button>
+      </td>
```
Comment (bellom) [inline src/components/Book.js:24]: Please we need to validate our props by doing the `Typechecking with PropTypes`, that was the mean reason you added it in milestone 1;
image

File: src/containers/BookForm.js [hunk 3]
```diff
@@ -10,18 +12,89 @@ const cat = [
...
+        return this.state;
+    }
+    return false;
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    if (this.state.title === "" || this.state.category === "") {
+      alert("Please enter your book name and category"); // eslint-disable-line no-alert
```
Comment (bellom) [inline src/containers/BookForm.js:50]: This should be production code, we shouldn't have `alert`, please remove the `alert`.

File: src/containers/BookForm.js [hunk 4]
```diff
@@ -10,18 +12,89 @@ const cat = [
...
+          ))}
+        </select>
+        <button type="submit" onClick={this.handleSubmit}>
+          Confirm
+        </button>
+      </form>
+    );
+  }
+}
+
```
Comment (bellom) [inline src/containers/BookForm.js:93]: Please we need to validate our props by doing the `Typechecking with PropTypes`, that was the mean reason you added it in milestone 1;
image

File: src/containers/BookList.js [hunk 5]
```diff
@@ -1,28 +1,51 @@
...
+              book={book}
+              clickHandler={this.handleRemoveBook}
+            />
+          ))}
+        </tbody>
+      </table>
+    );
+  }
+}
+
```
Comment (bellom) [inline src/containers/BookList.js:42]: Please we need to validate our props by doing the `Typechecking with PropTypes`, that was the mean reason you added it in milestone 1;
image

**Comment by Usagib** [comment]
Hello team, the comments on the last review were resolved
Submitting Pr for a new review.
Happy Coding!

**Pull request merged**